### ISSUE TITLE
Add form field for specifying phenotype namespace

### ISSUE THREAD

**Issue by jseager7** [pombase/canto#1463]
(Depends on issue  #1462)

Adding more phenotype annotation types should require changes to the user interface; something to allow the user to select a certain phenotype class before making a phenotype annotation.

I think the most obvious solution is an additional form field on the 'Search for phenotype terms' page, which the user can use to select a phenotype namespace before they can make an annotation. The form field could be:
- a dropdown menu, or
- radio buttons (since there are only three options).

To help curators who want to create many annotations in the same phenotype namespace at once, it might be useful to have the field remember the phenotype namespace that the user previously selected, rather than reverting back to a default (unless one namespace is expected to be far more commonly used than the others).

**Comment by jseager7** [comment]
Another option would be to present suggestions for all the phenotypes to the user (using auto-complete), then indicate which namespace the term belongs to afterwards, or even indicate the namespace while the user is typing the term.

**Comment by ValWood** [comment]
Hi @jseager7 ,

the plan is to  do:

![types](https://user-images.githubusercontent.com/7359272/40367693-ea866592-5dd1-11e8-8b99-c7004b66e074.jpg)
 

extending types to cover

host pathogen interaction phenotype
host phenotype
pathogen phenotype
(a bit like GO is one ontology with 3 name spaces and we split out the curation types)

We should revisit whether we still need to separate single and multi gene phenotypes. With  the new interface, we probably don't.....

**Comment by jseager7** [comment]
Hi @ValWood, thanks for that information. However, we also have the case when the annotation is made using the _genotype management_ screen, by following the link in the image below:

![genotype_management_annotation](https://user-images.githubusercontent.com/37659591/40414776-19a2a04e-5e71-11e8-9271-0fbe8de1dcd3.png)

And that leads to this screen:

![search_phenotype_term](https://user-images.githubusercontent.com/37659591/40414903-73330a36-5e71-11e8-96b5-9e3e2a4fcaa7.png)

This version of the workflow doesn't seem to support the other phenotype namespaces; the drop-down box only displays phenotypes from one namespace (_disease formation phenotype_).

Would the plan be to add the additional phenotype annotation types to the pop-up menu on the genotype table? (the one that appears to the right when you mouse-over a row of the table)

@kimrutherford might need your thoughts on the implementation side of this.

**Comment by ValWood** [comment]
Good point. But I we can always assume

multi organism phenotype = pathogen host interaction
host only single allele= host phenotype
pathogen only allele with host selected (host gene unknown)  = pathogen host interaction
pathogen only allele without host selected = pathogen phenotype

this would require that  the host  info is combined.
 If this isn't possible here, both options could be offered...

**Comment by jseager7** [comment]
I guess it would keep it simpler for the curators if we're able to infer which phenotype annotation to use – just as long as the phenotype annotations are exclusive enough that there's no risk of Canto inferring one type of annotation when the curator wanted another.

Maybe it would help to have some extra text when adding a phenotype, just confirming which type of phenotype annotation has been inferred… but maybe that would just confuse the curators.

A few more questions:

- Could you explain what you mean by the "host info [being] combined"? I'm not sure I follow.

- When you mention 'pathogen host interaction', does that correspond to the PHIPO namespace _disease formation phenotype_? I can see namespaces corresponding to _host phenotype_ and _pathogen phenotype_, but not the other one.

**Comment by kimrutherford** [comment]
> just as long as the phenotype annotations are exclusive enough that there's no risk of Canto
> inferring one type of annotation when the curator wanted another

I think it will work out OK.  There are three namespaces and three types of genotype (host, pathogen and host-pathogen-metagenotype).

> Maybe it would help to have some extra text when adding a phenotype, just confirming which type
> of phenotype annotation has been inferred

I think that's a good idea.  We should use the text "Host phenotype" instead of "Phenotype" for example.  You can test that by change "name: phenotype" to "name: host phenotype" in the configuration for the host phenotype ontology:

```yaml
available_annotation_type_list:
   ...
   - name: host phenotype
     category: ontology
     namespace: host_phenotype
     very_short_display_name: 'phenotype'
     short_display_name: 'host phenotype'
     display_name: 'host phenotype'
     synonyms_to_display:
   ...
```

(Sorry about all the repetition in the config - we can probably improve that)

**Comment by ValWood** [comment]
> Could you explain what you mean by the "host info [being] combined"? I'm not sure I follow.

I was thinking about  when you specify the host only when you have no host gene. I'm not sure how this will be linked up to be honest.


It would require that here:
https://docs.google.com/presentation/d/1UtWSxGFfU3KRZCMqmT6P3aFu94SZb-BxAe4gGmMDUdE/edit#slide=id.g3231978757_0_22

a pathogen gene is selected, and a host strain, to create a 'type' of multi-species genotype, but without host  genotype info (the genotype would be the WT strain genotype I suppose?)

> When you mention 'pathogen host interaction', does that correspond to the PHIPO namespace disease formation phenotype? I can see namespaces corresponding to host phenotype and pathogen phenotype, but not the other one.

Yes, sorry: old   'pathogen host interaction' , new  'disease formation phenotype'. I think this was renamed so it would not be confused with  'genetic' and 'physical interactions'.

**Comment by kimrutherford** [comment]
> I was thinking about when you specify the host only when you have no host gene. I'm not sure
> how this will be linked up to be honest.

I don't think this will be a problem.  If you're creating a pathogen-host meta-genotype you'll need pick the host and pathogen organisms from the dropdowns before you start adding alleles.  The selected host might have some genes in the session that can used to make alleles for the meta-genotype.  Or it might not.  Either way we know the host organism for the meta-genotype.

**Comment by ValWood** [comment]
Ok, so we should always know which type of phenotype based on the allele pathogen or host combination

**Comment by kimrutherford** [comment]
> Ok, so we should always know which type of phenotype based on the allele pathogen or host combination

Yep, I think so.

**Comment by jseager7** [comment]
So, to summarise:

- We don't need an extra form field for selecting phenotype annotation type on the _Genotype Management_ page, because it can be inferred from the meta-genotype.
- However, we should add some text on the _Search for phenotype term_ page that states which phenotype annotation type has been automatically inferred.

With regards to the single and multi-allele modes for phenotype annotations:

> We should revisit whether we still need to separate single and multi gene phenotypes. With the new interface, we probably don't.....

I've opened a new issue for this at #1477.

**Comment by jseager7** [comment]
I'm closing this issue for now, since I think I've added new issues to cover all of the things we discussed.
